pvlib/shading.py

```diff
@@ -553,3 +553,140 @@ def shaded_fraction1d(
     )
 
     return np.clip(t_asterisk, 0, 1)
+
+
+def direct_martinez(
+    poa_global,
+    poa_direct,
+    shaded_fraction,
+    shaded_blocks,
+    total_blocks,
+):
+    r"""
+    A shading losses power factor for non-monolithic Silicon
+    modules and arrays with an arbitrary number of bypass diodes.
+
+    This experimental model cancels a fraction of the direct and circumsolar
+    irradiance based on the number of *blocks* affected by the shadow.
+    More on blocks in the *Notes* section and in [1]_.
+
+    .. versionadded:: 0.11.0
+
+    Parameters
+    ----------
+    poa_global : numeric
+        Plane of array global irradiance. [W/m²].
+    poa_direct : numeric
+        Plane of array direct and circumsolar irradiance. [W/m²].
+    shaded_fraction : numeric
+        Fraction of module surface area that is shaded. Unitless.
+    shaded_blocks : numeric
+        Number of blocks affected by the shadow. Unitless.
+        If a floating point number is provided, it will be rounded up.
+    total_blocks : int
+        Number of total blocks. Unitless.
+
+    Returns
+    -------
+    shading_losses : numeric
+        Fraction of DC power lost due to shading. [unitless]
+
+    Notes
+    -----
+    The implemented equations are (6) and (8) from [1]_:
+
+    .. math::
+
+        (1 - F_{ES}) = (1 - F_{GS}) \left(1 - \frac{N_{SB}}{N_{TB} + 1}\right)
+        \quad \text{(6)}
+
+        \left(1 - \frac{P_{S}}{P_{NS}}\right) = \left(1 -
+        \frac{\left[(B + D^{CIR})(1 - F_{ES}) + D^{ISO} + R\right]}{G}\right)
+        \quad \text{(8)}
+
+    In (6), :math:`(1 - F_{ES})` is the correction factor to be multiplied by
+    the direct and circumsolar irradiance, :math:`F_{GS}` is the shaded
+    fraction of the collector, :math:`N_{SB}` is the number of shaded blocks
+    and :math:`N_{TB}` is the number of total blocks.
+
+    In (8), :math:`\frac{P_{S}}{P_{NS}}` is the fraction of DC power lost due
+    to shading, :math:`P_{S}` is the power output of the shaded module,
+    :math:`P_{NS}` is the power output of the non-shaded module,
+    :math:`B + D^{CIR}` is the beam and circumsolar irradiance,
+    :math:`D^{ISO} + R` is the sum of diffuse and albedo irradiances and
+    :math:`G` is the global irradiance.
+
+    **Blocks terminology:**
+
+    [1]_ defines a *block* as a group of solar cells protected by a bypass
+    diode. Also, a *block* is shaded when at least one of its cells is shaded.
```

Comment:
If one of its cells is "partly" shaded or "completely" shaded?

Reply:
> A better approximation is obtained by taking into account the shaded blocks. A ‘‘block’’ is here defined as a group of cells protected by one bypass diode. A block is shaded when at least one of its cells is shaded. A first possibility is to consider that the power of a block is fully cancelled when the block is shaded.

The paper does not specify it, but from the original tests spreadsheet I confirm they are considered shaded if a minimal amount of shade impacts on them.

Reply:
I've added *partially* to the docstring, thou maybe I shouldn't.

Reply:
I like it! Especially if you've confirmed with the spreadsheet